extensions/INTEL/SPV_INTEL_usm_storage_classes.asciidoc

```diff
@@ -0,0 +1,160 @@
+SPV_INTEL_usm_storage_classes
+=============================
+
+== Name Strings
+
+SPV_INTEL_usm_storage_classes
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+== Contributors
+
+- Joe Garvey, Intel
+
+== Notice
+
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+== Status
+
+Final Draft
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2020-04-30
+| Revision           | 1
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification,
+Version 1.6 Revision 2.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension introduces two new storage classes that are sub classes of the CrossWorkgroup storage class.  
+Using these more specific storage classes provides additional information that can enable optimization. 
+The extension also introduces two new conversion instructions to enable converting pointers from and to these storage classes.  
+
+== Extension Name
+To use this extension within a SPIR-V module, the following *OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_usm_storage_classes"
+----
+
+== New capabilities
+This extension introduces a new capability:
+
+----
+USMStorageClassesINTEL
+----
+
+== Token Number Assignments
+
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+| OpPtrCastToCrossWorkgroupINTEL | 5934
+| USMStorageClassesINTEL  | 5935
+| DeviceOnlyINTEL | 5936
+| HostOnlyINTEL | 5937
+| OpCrossWorkgroupCastToPtrINTEL | 5938
+|====
+
+== Modifications to the SPIR-V Specification, Version 1.6 Revision 2
+
+=== Storage Class
+
+Modify Section 3.7, Storage Class, adding these rows to the table:
+
+--
+[options="header"]
+|===
+2+^| Storage Class | Enabling Capabilities
+| 5936 | *DeviceOnlyINTEL* +
+A subset of the *CrossWorkgroup* storage class.  Memory that is resident on the device.  SYCL or OpenCL device unified shared memory. | *USMStorageClassesINTEL*
+| 5937 | *HostOnlyINTEL* +
+A subset of the *CrossWorkgroup* storage class.  Memory that is resident on the host.  SYCL or OpenCL host unified shared memory. | *USMStorageClassesINTEL*
+|===
+--
+
+=== Capability
+
+Modify Section 3.31, Capability, adding a row to the Capability table:
+--
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5935 | *USMStorageClassesINTEL* | *Kernel*
+|====
+--
+
+=== Instructions
+
+Modify Section 3.36.11, Conversion Instructions, adding two new instructions as follows:
+
+|===
+3+^| *OpPtrCastToCrossWorkgroupINTEL*
```

Comment:
This table renders OK with the GitHub asciidoc renderer but the HTML output is not ideal.  I think this will fix it.  Note, the change from `3+` to `4+` will also need the suggestion below, which changes `2+` to `1+`.

```suggestion
[cols="1,1,3*3",width="100%"]
|===
4+| *OpPtrCastToCrossWorkgroupINTEL*
```